packages/wrangler/templates/startDevWorker/ProxyWorker.ts

```diff
@@ -304,8 +304,23 @@ function insertLiveReloadScript(
  * so that this proxy is transparent to the Client Browser and User Worker.
  */
 function rewriteUrlRelatedHeaders(headers: Headers, from: URL, to: URL) {
+	const setCookie = headers.getAll("Set-Cookie");
+	headers.delete("Set-Cookie");
+	for (const cookie of setCookie) {
```

Comment:
NIT - if you put this for loop after the headers.forEach below then you wouldn't have to check that the key is not `set-cookie`.

Reply:
Happy to move it to after the `.forEach()`, but I think the check will still be required. The `.forEach()` calls `headers.set()` which AFAIK will clear all but the first `Set-Cookie` (while this section of code calls `headers.append()`)

Reply:
It shouldn't—this code should delete the original header, but then loop over the existing set-cookies to rewrite the domain